uflash.py

```diff
@@ -75,29 +77,32 @@ def find_microbit():
 
     Returns None if no micro:bit is found.
     """
+    # Check what sort of operating system we're on.
     if os.name == 'posix':
+        # 'posix' means we're on Linux or OSX (Mac).
+        # Call the unix "mount" command to list the mounted volumes.
         mount_output = check_output('mount').splitlines()
-        # (almost) list the mounted volumes.
         mounted_volumes = [x.split()[2] for x in mount_output]
-
         for volume in mounted_volumes:
-            if volume.endswith('MICROBIT'):
-                return volume
-
+            if volume.endswith(b'MICROBIT'):
+                return volume.decode('utf-8')  # Return a string not bytes.
     elif os.name == 'nt':
+        # 'nt' means we're on Windows.
         kernel32 = ctypes.windll.kernel32
 
-        vol_name_buf = ctypes.create_unicode_buffer(1024)
-        fs_name_buf = ctypes.create_unicode_buffer(1024)
-
-        serial_number = max_component_length = file_system_flags = None
-
-        def get_volume_name(volume):
+        def get_volume_name(disk_name):
             """
+            Each disk or external device connected to windows has an attribute
+            called "volume name". This function returns the volume name for
+            the given disk/device.
+
             Code from http://stackoverflow.com/a/12056414
             """
+            vol_name_buf = ctypes.create_unicode_buffer(1024)
+            fs_name_buf = ctypes.create_unicode_buffer(1024)
+            serial_number = max_component_length = file_system_flags = None
```

Comment:
I had also moved the buffer declarations (well, actually I've gotten rid of fs_name_buf after re-reading the kernel32 api docs) inside the function, at which point there's not really any reason for this not to be a top-level function